src/NServiceBus.Testing.Fakes/TestablePipelineContext.cs

```diff
@@ -21,6 +21,11 @@ public TestablePipelineContext(IMessageCreator messageCreator = null)
         }
 
         /// <summary>
+        /// A list of all messages sent with a timeout header.
+        /// </summary>
+        public TimeoutMessage<object>[] TimeoutMessages => timeoutMessages.ToArray();
```

Comment:
An alternative approach would be the following:

```csharp
public SentMessage<object>[] TimeoutMessages => sentMessages.Where(m => m.Options.GetHeaders().ContainsKey(Headers.IsSagaTimeoutMessage)).ToArray();
```

and then you just get the delay value from the `Options` property with `GetDeliveryDelay`/`GetDeliveryDate`.

With this approach, we don't need any other code changes, including the `TimeoutMessage` type.

There might even be an argument to be made that we don't need any of the changes here, and we just document the query above as the way to find timeout messages if you need them for a test.